routing/dht/routing.go

```diff
@@ -77,10 +79,23 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error
 	return nil
 }
 
+// ValReceipt represents a dht value record that has been received from a given peer
+// it is used to track peers with expired records in order to correct them.
+type ValReceipt struct {
+	From peer.ID
+	Val  *pb.Record
+}
```

Comment:
`receipts` in distributed protocols usually mean a message that the other side returns. in crypto protocols, receipts (and similar messages) often are message returned by a receiver that can be verified (e.g. has the hash of the message we sent, and a signature).  call it something else, because this name happens to collide with another meaning.

(Also without an actual receipt, we have no way of confirming the other side did indeed receive the record, so saying `"represents a dht value record that has been received from a given peer"` is not accurate, maybe `"represents a dht value record we recently put to a given peer"`)